moto/ec2/models.py

```diff
@@ -1415,6 +1429,19 @@ def authorize_security_group_egress(self,
                 if not is_valid_cidr(cidr):
                     raise InvalidCIDRSubnetError(cidr=cidr)
 
+        max_nb_rules = 50 if group.vpc_id else 100
```

Comment:
@spulec I've added a new commit. Have a look and let me know if you like the way I did it or not.
:)